onnxruntime_extensions/tools/add_HuggingFace_CLIPImageProcessor_to_model.py

```diff
@@ -0,0 +1,157 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+import argparse
+import os
+from pathlib import Path
+import onnx
+
+from .pre_post_processing import *
+
+
+class Dict2Class(object):
+    '''
+    Convert dict to class 
+    '''
+    def __init__(self, my_dict):
+        for key in my_dict:
+            setattr(self, key, my_dict[key])
+
+def image_processor(args: argparse.Namespace):
+    # support user providing encoded image bytes
+    steps = [
+        ConvertImageToBGR(),  # custom op to convert jpg/png to BGR (output is HWC)
+    ]  # Normalization params are for RGB ordering
+    if args.do_convert_rgb:
+        steps.append(ReverseAxis(axis=2, dim_value=3, name="BGR_to_RGB"))
+
+    if args.do_resize:
+        to_size = args.size
+        steps.append(Resize(to_size))
+
+    if args.do_center_crop:
+        to_size = args.crop_size
+        to_size = (to_size, to_size)
+        steps.append(CenterCrop(*to_size))
+
+    if args.do_rescale:
+        steps.append(ImageBytesToFloat(args.rescale_factor))
+    steps.append(ChannelsLastToChannelsFirst())
+
+    if args.do_normalize:
+        mean_std = list(zip(args.image_mean, args.image_std))
+        layout = 'CHW'
+        steps.append(Normalize(mean_std, layout=layout))
+
+    steps.append(Unsqueeze([0]))  # add batch dim
+
+    return steps
+
+
+def clip_image_processor(model_file: Path, output_file: Path, **kawrgs):
+    """
+    Used for models like stable-diffusion. should be compatible with 
+    https://github.com/huggingface/transformers/blob/main/src/transformers/models/clip/image_processing_clip.py
+
+    It applies to 'CLIP' image processor, and aligns with the HuggingFace class name.
+
+    A typical usage example is used in Stable diffusion.
+
+    :param model_file: The input model file path.
+    :param output_file: The output file path, where the finalized model saved to.
+    :param onnx_opset: The opset version of onnx model, default(16).
+    :param extend_config: The extended config for the model.
+    """
+    args = Dict2Class(kawrgs)
```

Comment:
What's the benefit of this vs just passing the args from argparse directly? The usage seems the same either way.

Reply:
It supports the usage like `clip_image_processor(param=value,.....)`. Otherwise, we have to pass a data object.